optimum/deepsparse/modeling_diffusion.py

```diff
@@ -0,0 +1,527 @@
+
+        return cls(
+            vae_decoder_path=vae_decoder_path,
+            text_encoder_path=text_encoder_path,
+            unet_path=unet_path,
+            config=config,
+            tokenizer=sub_models.get("tokenizer", None),
+            scheduler=sub_models.get("scheduler"),
+            feature_extractor=sub_models.get("feature_extractor", None),
+            tokenizer_2=sub_models.get("tokenizer_2", None),
+            vae_encoder_path=vae_encoder_path,
+            text_encoder_2_path=text_encoder_2_path,
+            model_save_dir=Path(model_save_dir.name),
+        )
+
+    @classmethod
+    def _from_transformers(
+        cls,
+        model_id: str,
+        config: Optional[str] = None,
+        use_auth_token: Optional[Union[bool, str]] = None,
+        revision: str = "main",
+        force_download: bool = True,
+        cache_dir: Optional[str] = None,
+        subfolder: str = "",
+        local_files_only: bool = False,
+        trust_remote_code: bool = False,
+        task: Optional[str] = None,
+    ):
+        if task is None:
+            task = cls._auto_model_to_task(cls.auto_model_class)
+
+        save_dir = Path("tmp")
+        save_dir_path = Path(save_dir.name)
+
+        main_export(
+            model_name_or_path=model_id,
+            output=save_dir_path,
+            task=task,
+            do_validation=False,
+            no_post_process=True,
+            subfolder=subfolder,
+            revision=revision,
+            cache_dir=cache_dir,
+            use_auth_token=use_auth_token,
+            local_files_only=local_files_only,
+            force_download=force_download,
+            trust_remote_code=trust_remote_code,
+        )
+
+        return cls._from_pretrained(
+            save_dir_path,
+            config=config,
+            model_save_dir=save_dir,
+        )
+
+    @classmethod
+    def _load_config(cls, config_name_or_path: Union[str, os.PathLike], **kwargs):
+        return cls.load_config(config_name_or_path, **kwargs)
+
+    def _save_config(self, save_directory):
+        self.save_config(save_directory)
+
+    def to(self, device: str):
+        self._device = device.upper()
+        return self
+
+    @property
+    def device(self) -> str:
+        return self._device.lower()
+
+
+# TODO : Use ORTModelPart once IOBinding support is added
+class _DeepSparseDiffusionModelPart:
+    """
+    For multi-file ONNX models, represents a part of the model.
+    """
+
+    CONFIG_NAME = "config.json"
+
+    def __init__(self, engine: deepsparse.Engine, parent_model: DeepSparseBaseModel, model_name: str = "encoder"):
+        self.engine = engine
+        self.parent_model = parent_model
+        self._model_name = model_name
+        config_path = parent_model.model_save_dir / model_name / self.CONFIG_NAME
+        self.config = self.parent_model._dict_from_json_file(config_path) if config_path.is_file() else {}
+        self.input_dtype = {"latent_sample": np.float64}
```

Comment:
does this need to be in the base class? i don't see it used

Reply:
Which line exactly are you referring to here? I am certain line 418 will throw an error if removed and line 417 is used to get the model configuration.